Allow to select a slot (e.g. scaled, raw etc..) in join_features()

noriakis opened the pull request:
@stemangiola 

I have made the following changes to `join_features()` and checked that `R CMD CHECK` does not produce any warnings or errors. I would be grateful if you could review the PR.

- add params for `assay` `slot` and `prefix` in `get_abundance_sc_wide()`
- make `get_abundance_sc_long()` handle `assay` and `slot` specification
- add arguments and params for `assay` `slot` in `get_abundance_sc_long()`
- add params for `assay` `slot` and `...` in `join_features()`

stemangiola reviewed (changes requested):
Thanks! Can you please add unit tests, similar to join_features wide?

stemangiola commented on R/utilities.R:
```diff
@@ -154,11 +156,13 @@ get_abundance_sc_wide = function(.data, features = NULL, all = FALSE, assay = .d
 #'
 #' @export
-get_abundance_sc_long = function(.data, features = NULL, all = FALSE, exclude_zeros = FALSE){
+get_abundance_sc_long = function(.data, features = NULL, all = FALSE, exclude_zeros = FALSE, assay = .data@active.assay, slot = "data"){
```
assay here is not needed as we will return all assays (inside the function is clear).

stemangiola commented on R/utilities.R:
```diff
@@ -187,20 +191,24 @@ get_abundance_sc_long = function(.data, features = NULL, all = FALSE, exclude_ze
 
   # Else
   else variable_genes = NULL
+  
+  DefaultAssay(.data) = assay
+  for(i in Assays(.data) %>% setdiff(assay)) {
+    .data[[i]] = NULL
+  }
 
```
This is not needed as we return all assays

noriakis commented:
Thank you very much for your prompt review. I apologize for the misinterpretation of the feature. I have removed the `assay` argument from `get_abundance_sc_long`, and added a unit test for join_feature wide and long.

stemangiola commented on R/methods.R:
```diff
@@ -98,7 +99,9 @@ setMethod("join_features", "Seurat",  function(.data,
                                                exclude_zeros = FALSE,
-                                               shape = "long", ...)
+                                               shape = "long",
+                                               assay = .data@active.assay, 
```
OK I see now that we can find a way in between.

1) `join_feature` can have assay = NULL by default

2) `get_abundance_sc_wide` can have `assay = Assays(pbmc_small)` by default, and inside the function you can use `assay` instead of `assay_names`.

3) `get_abundance_sc_long` can have assay = .data@active.assay by default

noriakis replied:
Thank you very much for your guidance. I set `join_feature` to NULL by default, and set default values for `get_abundance_sc_long` and `get_abundance_sc_wide`. I apologize if I am missing something.
